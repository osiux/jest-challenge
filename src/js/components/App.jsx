import React from "react";

import Counter from "./Counter";
import Controls from "./Controls";

class App extends React.Component {
    constructor() {
        super();

        this.state = {
            counter: 0
        };
    }

    _increaseCounter = () =>
        this.setState(prevState => {
            return {
                counter: prevState.counter + 1
            };
        });

    _decreaseCounter = () =>
        this.setState(prevState => {
            return {
                counter: prevState.counter - 1
            };
        });

    _resetCounter = () =>
        this.setState({
            counter: 0
        });

    render() {
        const { counter } = this.state;

        return (
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col-sm-12 col-md-6">
                        <h1 className="text-center">Counter</h1>

                        <div className="card text-center">
                            <Counter counter={counter} />

                            <Controls
                                onDecrease={this._decreaseCounter}
                                onReset={this._resetCounter}
                                onIncrease={this._increaseCounter}
                            />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default App;
