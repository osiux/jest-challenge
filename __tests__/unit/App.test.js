import React from 'react';
import { shallow } from 'enzyme';

import App from '../../src/js/components/App';

describe('App component', () => {
    it('matches snapshot', () => {
        const component = shallow(<App />);

        expect(component).toMatchSnapshot();
    });

    it('renders Counter and Controls components', () => {
        const component = shallow(<App />);

        expect(component.find('Counter').exists()).toBe(true);
        expect(component.find('Controls').exists()).toBe(true);
    });
});
