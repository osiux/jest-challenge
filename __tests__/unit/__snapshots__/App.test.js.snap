// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-sm-12 col-md-6"
    >
      <h1
        className="text-center"
      >
        Counter
      </h1>
      <div
        className="card text-center"
      >
        <Counter
          counter={0}
        />
        <Controls
          onDecrease={[Function]}
          onIncrease={[Function]}
          onReset={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
